morph_tool/morphdb.py

```diff
@@ -0,0 +1,257 @@
+'''Provides two classes to work with neurondb files
+
+The main use is to provide a neuronDB loader and a method to retrieve
+information as a dataframe:
+
+MorphologyDB('neurondb.xml').df
+'''
+import json
+import logging
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterable, List, Optional
+
+import pandas as pd
+import xmltodict
+
+L = logging.getLogger(__name__)
+
+NEURONDB_XML = 'neuronDB.xml'
+MTYPE_MSUBTYPE_SEPARATOR = ':'
+
+
+class MorphInfo:
+    '''A class the contains information about a morphology.
+
+    Its role is to abstract away the raw data.
+    '''
+
+    # Attributes found in the <repair></repair> block of the XML
+    BOOLEAN_REPAIR_ATTRS = [
+        'use_axon',
+        'use_dendrites',
+        'axon_repair',
+        'dendrite_repair',
+        'basal_dendrite_repair',
+        'tuft_dendrite_repair',
+        'oblique_dendrite_repair',
+        'unravel',
+        'use_for_stats',
+    ]
+
+    COLUMNS = ['name', 'mtype', 'msubtype', 'fullmtype',
+               'layer'] + BOOLEAN_REPAIR_ATTRS + ['axon_inputs']
+
+    def __init__(self, item: Dict[str, Any]):
+        '''MorphInfo ctor.
+
+        Args:
+            item: A dictionnary that represents the content of the XML file.
+                  The only mandatory keys are [name, mtype, layer]
+        '''
+        self.item = item
```

Comment:
It looks to me like it's easy to get things out of sync here: if someone assigns to `self.name`, the `self.item` dict isn't updated

Reply:
True. Well, I think we can actually discard this key. There is no real reason to keep the input data inside this class.